arch/arm/core/cortex_m/vector_table.S

```diff
@@ -66,5 +66,9 @@ SECTION_SUBSEC_FUNC(exc_vector_table,_vector_table_section,_vector_table)
 #endif /* CONFIG_ARMV6_M_ARMV8_M_BASELINE */
     .word __reserved
     .word __pendsv
+#if defined(CONFIG_SYS_CLOCK_EXISTS)
```

Comment:
I am not sure what is the best to do here

Reply:
I mean, this is clearly only for the case when the SysTick is the HW timer providng system clock, right? Would we need to #ifdef out other modules providing timing, if CONFIG_SYS_CLOCK_EXISTS is not defined?

Reply:
well, it works? I mean, the sample boots properly and no code related to timers is included. This Kconfig option has long existed as a way to remove timer support for the kernel. I really don't think we need to do anything else. If another module requires timing, it needs to `depend` on `SYS_CLOCK_EXISTS` or `select` it